File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0014-loan-comparison/type/TMetric.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tMetric"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TMetric extends com.gs.dmn.runtime.DMNType {
    static TMetric toTMetric(Object other) {
        if (other == null) {
            return null;
        } else if (TMetric.class.isAssignableFrom(other.getClass())) {
            return (TMetric)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TMetricImpl result_ = new TMetricImpl();
            result_.setLenderName((String)((com.gs.dmn.runtime.Context)other).get("lenderName"));
            result_.setRate((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("rate"));
            result_.setPoints((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("points"));
            result_.setFee((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("fee"));
            result_.setLoanAmt((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("loanAmt"));
            result_.setDownPmtAmt((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("downPmtAmt"));
            result_.setPaymentAmt((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("paymentAmt"));
            result_.setEquity36moPct((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("equity36moPct"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTMetric(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TMetric.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("lenderName")
    String getLenderName();

    @com.fasterxml.jackson.annotation.JsonGetter("rate")
    java.math.BigDecimal getRate();

    @com.fasterxml.jackson.annotation.JsonGetter("points")
    java.math.BigDecimal getPoints();

    @com.fasterxml.jackson.annotation.JsonGetter("fee")
    java.math.BigDecimal getFee();

    @com.fasterxml.jackson.annotation.JsonGetter("loanAmt")
    java.math.BigDecimal getLoanAmt();

    @com.fasterxml.jackson.annotation.JsonGetter("downPmtAmt")
    java.math.BigDecimal getDownPmtAmt();

    @com.fasterxml.jackson.annotation.JsonGetter("paymentAmt")
    java.math.BigDecimal getPaymentAmt();

    @com.fasterxml.jackson.annotation.JsonGetter("equity36moPct")
    java.math.BigDecimal getEquity36moPct();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("lenderName", getLenderName());
        context.put("rate", getRate());
        context.put("points", getPoints());
        context.put("fee", getFee());
        context.put("loanAmt", getLoanAmt());
        context.put("downPmtAmt", getDownPmtAmt());
        context.put("paymentAmt", getPaymentAmt());
        context.put("equity36moPct", getEquity36moPct());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TMetric other = (TMetric) o;
        if (this.getDownPmtAmt() != null ? !this.getDownPmtAmt().equals(other.getDownPmtAmt()) : other.getDownPmtAmt() != null) return false;
        if (this.getEquity36moPct() != null ? !this.getEquity36moPct().equals(other.getEquity36moPct()) : other.getEquity36moPct() != null) return false;
        if (this.getFee() != null ? !this.getFee().equals(other.getFee()) : other.getFee() != null) return false;
        if (this.getLenderName() != null ? !this.getLenderName().equals(other.getLenderName()) : other.getLenderName() != null) return false;
        if (this.getLoanAmt() != null ? !this.getLoanAmt().equals(other.getLoanAmt()) : other.getLoanAmt() != null) return false;
        if (this.getPaymentAmt() != null ? !this.getPaymentAmt().equals(other.getPaymentAmt()) : other.getPaymentAmt() != null) return false;
        if (this.getPoints() != null ? !this.getPoints().equals(other.getPoints()) : other.getPoints() != null) return false;
        if (this.getRate() != null ? !this.getRate().equals(other.getRate()) : other.getRate() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getDownPmtAmt() != null ? this.getDownPmtAmt().hashCode() : 0);
        result = 31 * result + (this.getEquity36moPct() != null ? this.getEquity36moPct().hashCode() : 0);
        result = 31 * result + (this.getFee() != null ? this.getFee().hashCode() : 0);
        result = 31 * result + (this.getLenderName() != null ? this.getLenderName().hashCode() : 0);
        result = 31 * result + (this.getLoanAmt() != null ? this.getLoanAmt().hashCode() : 0);
        result = 31 * result + (this.getPaymentAmt() != null ? this.getPaymentAmt().hashCode() : 0);
        result = 31 * result + (this.getPoints() != null ? this.getPoints().hashCode() : 0);
        result = 31 * result + (this.getRate() != null ? this.getRate().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("downPmtAmt=" + getDownPmtAmt());
        result_.append(", equity36moPct=" + getEquity36moPct());
        result_.append(", fee=" + getFee());
        result_.append(", lenderName=" + getLenderName());
        result_.append(", loanAmt=" + getLoanAmt());
        result_.append(", paymentAmt=" + getPaymentAmt());
        result_.append(", points=" + getPoints());
        result_.append(", rate=" + getRate());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0014-loan-comparison/type/TMetric.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tMetric"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TMetricImpl::class)
interface TMetric : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("lenderName")
    val lenderName: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("rate")
    val rate: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("points")
    val points: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("fee")
    val fee: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("loanAmt")
    val loanAmt: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("downPmtAmt")
    val downPmtAmt: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("paymentAmt")
    val paymentAmt: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("equity36moPct")
    val equity36moPct: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("lenderName", this.lenderName)
        context.put("rate", this.rate)
        context.put("points", this.points)
        context.put("fee", this.fee)
        context.put("loanAmt", this.loanAmt)
        context.put("downPmtAmt", this.downPmtAmt)
        context.put("paymentAmt", this.paymentAmt)
        context.put("equity36moPct", this.equity36moPct)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TMetric
        if (if (this.downPmtAmt != null) this.downPmtAmt != other.downPmtAmt else other.downPmtAmt != null) return false
        if (if (this.equity36moPct != null) this.equity36moPct != other.equity36moPct else other.equity36moPct != null) return false
        if (if (this.fee != null) this.fee != other.fee else other.fee != null) return false
        if (if (this.lenderName != null) this.lenderName != other.lenderName else other.lenderName != null) return false
        if (if (this.loanAmt != null) this.loanAmt != other.loanAmt else other.loanAmt != null) return false
        if (if (this.paymentAmt != null) this.paymentAmt != other.paymentAmt else other.paymentAmt != null) return false
        if (if (this.points != null) this.points != other.points else other.points != null) return false
        if (if (this.rate != null) this.rate != other.rate else other.rate != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.downPmtAmt != null) this.downPmtAmt.hashCode() else 0)
        result = 31 * result + (if (this.equity36moPct != null) this.equity36moPct.hashCode() else 0)
        result = 31 * result + (if (this.fee != null) this.fee.hashCode() else 0)
        result = 31 * result + (if (this.lenderName != null) this.lenderName.hashCode() else 0)
        result = 31 * result + (if (this.loanAmt != null) this.loanAmt.hashCode() else 0)
        result = 31 * result + (if (this.paymentAmt != null) this.paymentAmt.hashCode() else 0)
        result = 31 * result + (if (this.points != null) this.points.hashCode() else 0)
        result = 31 * result + (if (this.rate != null) this.rate.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("downPmtAmt=" + downPmtAmt)
        result_.append(", equity36moPct=" + equity36moPct)
        result_.append(", fee=" + fee)
        result_.append(", lenderName=" + lenderName)
        result_.append(", loanAmt=" + loanAmt)
        result_.append(", paymentAmt=" + paymentAmt)
        result_.append(", points=" + points)
        result_.append(", rate=" + rate)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTMetric(other: Any?): TMetric? {
            if (other == null) {
                return null
            } else if (other is TMetric?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TMetricImpl()
                result_.lenderName = other.get("lenderName") as String?
                result_.rate = other.get("rate") as java.math.BigDecimal?
                result_.points = other.get("points") as java.math.BigDecimal?
                result_.fee = other.get("fee") as java.math.BigDecimal?
                result_.loanAmt = other.get("loanAmt") as java.math.BigDecimal?
                result_.downPmtAmt = other.get("downPmtAmt") as java.math.BigDecimal?
                result_.paymentAmt = other.get("paymentAmt") as java.math.BigDecimal?
                result_.equity36moPct = other.get("equity36moPct") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTMetric(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TMetric::class.java.getSimpleName()))
            }
        }
    }
}
